app/models/framework/management_charge_calculator/flat_rate.rb

```diff
@@ -34,7 +36,7 @@ def overridding_calculation_column?
 
       def cast_to_decimal(column_value)
         case column_value
-        when String then BigDecimal(column_value)
+        when String then BigDecimal(column_value.gsub(NUMBER_WHITELIST, ''))
```

Comment:
This is where the naming made me think we were substituting the accepted numerical values with nothing, and keeping everything else...